### PR TITLE
test/e2e/network: Honor --dns-domain in more places

### PR DESCRIPTION
**Pull request by towolf**
Try to finish what commit 4c8a65ac014aac286dc081db409edf800b18bf8f started; that is, do not assume
cluster.local is a constant base domain, when it is configurable.

This makes DNS e2e tests pass with --dns-domain, which was only being honored
for some tests, not all. I hope I understood the purpose of this flag correctly.

Signed-off-by: Tobias Wolf <towolf@gmail.com>

**What type of PR is this?**

/kind failing-test

### CODE REVIEW

**Comment by towolf** [comment]
@nagiesek, PTAL?

**Comment by towolf** [comment]
@rajansandeep isn;t this along the lines of your earlier 4c8a65ac014aac286dc081db409edf800b18bf8f?

There are more tests that hard-code `cluster.local` now.

File: test/e2e/network/dns.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ import (
 	"strings"
 	"time"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (rajansandeep) [inline test/e2e/network/dns.go]: Seems like an unnecessary change.

**Comment by rajansandeep** [comment]
/cc @MrHohn 
ptal?

File: test/e2e/network/dns.go (hunk 1 above)
Comment (towolf) [reply test/e2e/network/dns.go]: Removed, my editor did this behind my back ...

**Review by MrHohn — APPROVED** [review]
Thanks!
/lgtm
/approve

**Comment by vryzhenkin** [comment]
Hey! It's possible to cherry-pick it to release-1.14?

**Comment by MrHohn** [comment]
@vryzhenkin What would be the reason for cherrypicking?

**Comment by vryzhenkin** [comment]
@MrHohn Only to make this test pass on 1.14 clusters with differently configured domains...

**Comment by MrHohn** [comment]
SGTM, would you mind opening up the cherrypick and assign it to me?

**Comment by vryzhenkin** [comment]
@MrHohn Sure! I'll do it! Thank you!

**Comment by towolf** [comment]
We are passing the Conformance suite now, thanks for merging!
image
